js/validation/primitives.js

```diff
@@ -29,7 +29,7 @@ function validateUrls(fieldName, doc) {
 
 // Validates a single URL
 function validateUrl(fieldName, doc) {
-    if (!isUrl(doc)) {
+    if (!isBlankNodeId(doc) && !isUrl(doc)) {
```

Comment:
You probably need to turn this into two functions: one for `@id` (which may be blank node identifiers or URIs) and one for actual URLs.